imap_processing/codice/L0/lo_pha.py

```diff
@@ -0,0 +1,220 @@
+"""This is the xtce generator for the P_COD_LO_PHA packet."""
+
+import xml.etree.ElementTree as Et
+
+import pandas as pd
+
+from tools.xtce_generation.ccsds_header_xtce_generator import CCSDSParameters
+
+"""
+Important Steps:
+- packet_name: This is the name of the packet (sheet) in the Excel file. This is
+case-sensitive.
+- path_to_excel_file: This is the path to the Excel file with packet definitions
+- ccsds_parameters: This is a list of dictionaries containing the CCSDS parameters
+- APID: This is the APID of the packet. This is used to filter the parameters
+in the CoDICE Science Packet.
+- sci_byte (optional): This is the BYTE number of lenghtInBits in the "Data" mnemonic
+in the CoDICE Science Packet.
+
+"""
+
+packet_name = "P_COD_LO_PHA"
+
+path_to_excel_file = "/Users/gamo6782/Desktop/IMAP/TLM_COD_20230629-110638(update).xlsx"
+
+# Load CCSDS parameters from the CCSDSParameters class
+ccsds_parameters = CCSDSParameters().parameters
+
+if __name__ == "__main__":
+    # ET.register_namespace is important!
+    # This is the namespace for the IMAP xtce files currently.
+    Et.register_namespace(
+        "xtce", "http://www.omg.org/space"
+    )  # Update the namespace here
+
+    # Load data from Excel file
+    xls = pd.ExcelFile(path_to_excel_file)
+    pkt = xls.parse(packet_name)
+
+    # Input the correct APID here
+    APID = "1153"
+
+    # sci_byte is the BYTE number of lenghtInBits in the "Data" mnemonic
+    sci_byte = 276480
+
+    # Fill missing values with '-*-'
+    pkt.fillna("", inplace=True)
+
+    # Create the root element and add namespaces
+    root = Et.Element("{http://www.omg.org/space}SpaceSystem")
+
+    root.attrib["name"] = packet_name
+
+    # Create the Header element with attributes 'date', 'version', and 'author'
+    # Versioning is used to keep track of changes to the XML file.
+    header = Et.SubElement(root, "{http://www.omg.org/space}Header")
+    header.attrib["date"] = "2023-09"
+    header.attrib["version"] = "1.0"
+    header.attrib["author"] = "IMAP SDC"
+
+    """
+    These lines of code create a structured XML hierarchy to represent data according
+    to a defined XML schema:
+    - The `TelemetryMetaData` element is added under the 'root' element,
+    providing a namespace.
+    - Within `TelemetryMetaData`, the `ParameterTypeSet` element is created
+    to define parameter types.
+    - Similarly, the `ParameterSet` element is added to `TelemetryMetaData`
+    to define specific parameters.
+    These elements and their organization help organize and standardize
+    data representation in XML format.
+    """
+
+    # Create the TelemetryMetaData element
+    telemetry_metadata = Et.SubElement(
```

Comment:
This looks like a fair amount of duplicated code - could you pull out parts of this into methods, so you can reuse it in all these processing steps? Maybe by passing in the unique names as a dictionary?

Reply:
From running `hi_pha` and `lo_nsw` in diffchecker, it looks like they are almost identical. Definitely pull that into a function so you can reuse it.